### PR TITLE
build(ts): Upgrade to typescript@3.7.2

### PR DESCRIPTION
**Pull request by billyvg**
New Features:

* Optional Chaining
* Nullish Coalescing
* Assertion Functions
* Better Support for never-Returning Functions
* --declaration and --allowJs
* (More) Recursive Type Aliases
* The useDefineForClassFields Flag and The declare Property Modifier
* Build-Free Editing with Project References
* Uncalled Function Checks
* Flatter Error Reporting
* // @ts-nocheck in TypeScript Files
* Semicolon Formatter Option
* Website and Playground Updates
* Breaking Changes
* DOM Changes
* Class Field Mitigations
* Function Truthy Checks
* Local and Imported Type Declarations Now Conflict
* API Changes

[Read release announcement](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/)

### CODE REVIEW

**Comment by dashed** [comment]
Sweet. 🎉

File: src/sentry/static/sentry/app/components/events/interfaces/spans/dividerHandlerManager.tsx [hunk 1]
```diff
@@ -173,7 +173,7 @@ export class Provider extends React.Component<PropType, StateType> {
     this.isDragging = false;
 
     selectRefs(this.dividerLineRefs, (dividerDOM: HTMLDivElement) => {
-      dividerDOM.style.backgroundColor = null;
+      dividerDOM.style.backgroundColor = '';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/interfaces/spans/dividerHandlerManager.tsx]: cc @dashed - this needs to be a string

File: src/sentry/static/sentry/app/components/links/externalLink.tsx [hunk 2]
```diff
@@ -1,37 +1,8 @@
...
-};
-
-// Should we allow these to be overridden?
-ExternalLink.defaultProps = {
-  target: '_blank',
-  rel: 'noreferrer noopener',
-};
-
-export default ExternalLink;
+export default React.forwardRef<HTMLAnchorElement, Props>((props, ref) => (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/links/externalLink.tsx]: cc @EvanPurkhiser - TS@3.7 was unhappy with that we had before, I think we can just simplify it to this

File: src/sentry/static/sentry/app/components/events/interfaces/spans/dividerHandlerManager.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/interfaces/spans/dividerHandlerManager.tsx]: Whoops.

File: src/sentry/static/sentry/app/views/discover/styles.tsx [hunk 3]
```diff
@@ -62,7 +62,7 @@ export const ResultViewActions = styled('div')`
   }
 `;
 
-export const ResultViewButtons = styled(NavTabs)<NavTabs['props']>`
+export const ResultViewButtons = styled(NavTabs)`
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/discover/styles.tsx]: @billyvg are we going away from this preference of props for emotion styled components in TS 3.7?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/discover/styles.tsx]: This is invalid in 3.7 - 
```
21:43:09 webpack | ERROR in /Users/billy/Dev/sentry/src/sentry/static/sentry/app/views/incidents/details/body.tsx
21:43:09 webpack | ./app/views/incidents/details/body.tsx
21:43:09 webpack | [tsl] ERROR in /Users/billy/Dev/sentry/src/sentry/static/sentry/app/views/incidents/details/body.tsx(130,39)
21:43:09 webpack |       TS2709: Cannot use namespace 'NavTabs' as a type.

```

It also: 1) does not reference any NavTabs props in the styling and 2) can infer the component's props so is not needed.

Comment (dashed) [reply src/sentry/static/sentry/app/views/discover/styles.tsx]: gotcha

**Review by dashed — APPROVED** [review]
Excited for TS v3.7

**Pull request merged**